saleor/graphql/order/mutations/draft_order_update.py

```diff
@@ -63,48 +98,211 @@ def should_invalidate_prices(cls, cleaned_input, *args) -> bool:
         )
 
     @classmethod
-    def save(cls, info: ResolveInfo, instance, cleaned_input):
+    def clean_input(
+        cls, info: ResolveInfo, instance: models.Order, data: dict, **kwargs
+    ):
+        cls.clean_channel_id(instance, data)
         manager = get_plugin_manager_promise(info.context).get()
-        app = get_app_promise(info.context).get()
-        return cls._save_draft_order(
-            info,
-            instance,
-            cleaned_input,
-            is_new_instance=False,
-            app=app,
-            manager=manager,
+        shipping_address = data.pop("shipping_address", None)
+        billing_address = data.pop("billing_address", None)
+        redirect_url = data.pop("redirect_url", None)
+
+        shipping_method_input = {}
+        if "shipping_method" in data:
+            shipping_method_input["shipping_method"] = get_shipping_model_by_object_id(
+                object_id=data.pop("shipping_method", None),
+                error_field="shipping_method",
+            )
+
+        if email := data.get("user_email", None):
+            try:
+                user = User.objects.get(email=email, is_active=True)
+                data["user"] = graphene.Node.to_global_id("User", user.id)
+            except User.DoesNotExist:
+                data["user"] = None
+
+        cleaned_input = super().clean_input(info, instance, data, **kwargs)
+
+        cleaned_input.update(shipping_method_input)
+
+        channel = instance.channel or cleaned_input.get("channel_id")
+        DraftOrderCleaner.clean_voucher_and_voucher_code(channel, cleaned_input)
+
+        cls.clean_addresses(
+            info, instance, cleaned_input, shipping_address, billing_address, manager
         )
 
+        DraftOrderCleaner.clean_redirect_url(redirect_url, cleaned_input)
+
+        return cleaned_input
+
+    @classmethod
+    def clean_channel_id(cls, instance, data):
+        if data.get("channel_id"):
+            if hasattr(instance, "channel"):
+                raise ValidationError(
+                    {
+                        "channel_id": ValidationError(
+                            "Can't update existing order channel id.",
+                            code=OrderErrorCode.NOT_EDITABLE.value,
+                        )
+                    }
+                )
+
+    @classmethod
+    def clean_addresses(
```

Comment:
I do not move `clean_address` as it requires `I18nMixin` and `ModelMutation` to work.